internal/pkg/siderolink/loghandler.go

```diff
@@ -68,24 +72,35 @@ func (h *LogHandler) Start(ctx context.Context) error {
 		return err
 	}
 
-	var tickerCh <-chan time.Time
+	nextFlushDuration := func() time.Duration {
+		if h.StorageFlushJitter == 0 {
+			return h.StorageFlushPeriod
+		}
+
+		jitter := time.Duration(rand.Int64N(int64(h.StorageFlushJitter)*2) - int64(h.StorageFlushJitter))
 
-	var storagePath string
+		return h.StorageFlushPeriod + jitter
+	}
 
-	storage, storageEnabled := h.Cache.Storage.Get()
-	if storageEnabled {
-		ticker := time.NewTicker(h.StorageFlushPeriod)
-		tickerCh = ticker.C
+	var (
+		timer       *time.Timer
+		timerCh     <-chan time.Time
+		storagePath string
+	)
 
-		defer ticker.Stop()
+	storage, ok := h.Cache.Storage.Get()
+	if ok {
+		timer = time.NewTimer(nextFlushDuration())
+		defer timer.Stop()
```

Comment:
you might also need to drain the timer when you stop it:
https://pkg.go.dev/time#Timer.Stop